website/sandbox.ts

```
@@ -110,3 +110,10 @@ matplotlib.setOutputHandler({
     rpc.call("imshow", guessCellId(), data);
   }
 });
+
+window.addEventListener("error", (ev: ErrorEvent) => {
+  const error = ev.error;
+  const cellId = guessCellId(error);
+  const stack = transpiler.formatError(error);
+  rpc.call("console", cellId, stack);
+});
```

What causes this error event? Is this part of the iframe API?

---

window.onerror received "uncaught" exceptions. Let's say you did this in a notebook:

```
setTimeout(() => {
  throw new Error('err')
}, 100);
```

Currently the error would be swallowed. This makes sure it gets shown to the user; we're guessing the cell originating as well as we can.

I see there is a bug in it though, will fix soon.

---

Maybe add a comment about this here with a TODO.

---

I fixed the bug